middleman-core/lib/middleman-core/preview_server.rb

```diff
@@ -112,12 +112,13 @@ def new_app
 
           config[:environment] = opts[:environment].to_sym if opts[:environment]
           config[:port] = opts[:port] if opts[:port]
+          config[:host] = opts[:host].to_s.empty? ? Socket.gethostname.tr(' ', '+') : opts[:host]
```

Comment:
`opts[:host]` will be `nil` if it's not set, but if you want to be extra careful, you can write `opts[:host].presence || Socket.gethostname.tr(' ', '+')`. See [Object#presence](http://api.rubyonrails.org/classes/Object.html#method-i-presence).